src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,158 @@ def test_set_beam_shift(self):
         self.scanner.beamShift.value = (new_beam_shift_x, new_beam_shift_y)
         self.assertAlmostEqual((new_beam_shift_x, new_beam_shift_y), self.scanner.beamShift.value)
 
+    @unittest.skip("Currently this functionality is not supported by the XT client and the code is commented out")
+    def test_apply_autostigmator(self):
+        """
+        Test for the auto functionality of the apply_autostigmator.
+        Runs the apply_autostigmator multiple times and checks (using "get_stigmator()) if the found stigmator values
+        are equal each run. Checks the returned state, the return object and if an error is raised for an incorrect input state.
+        """
+        detector = 'se-detector'  # Test autostigmator only for se-detector, channel=electron1
+        ## Basic testing of the functionality of the wrapper: can it be started and canceled
+        autostigmator_return = self.scanner.applyAutoStigmator(detector)
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, True)
+        self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+
+        autostigmator_return.cancel()
+        time.sleep(5.0)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, False)
+
+        ## Test if autostigmating gives repeatable results and check time it takes to auto stigmate.
+
+        # Create sets holding stigmator values to check if the outcome is consistent
+        stigmator_values_x = []  # List holding only the stigmator values in x direction
+        stigmator_values_y = []  # List holding only the stigmator values in y direction
+
+        # Run apply_autostigmator multiple times and afterwards check if the outcome is the same for each run
```

Comment:
Can you actually be sure that you will find exactly the same values whenever you run auto stigmation? Maybe check this with Wilco or Job.